app/uk/gov/hmrc/nrs/retrieval/config/HttpHead.scala

```diff
@@ -42,46 +43,47 @@ object CoreHttpReads extends HttpErrorFunctions {
 
   def responseHandler(method: String, url: String, response: HttpResponse): HttpResponse = {
     response.status match {
-      case status if (status == 404) => {
+      case status if status == NOT_FOUND =>
         logger.info(s"Submission bundle not found $status for query $method $url")
         if(method == HEAD_VERB) {
           response
         } else {
-          HttpResponse(404, JsArray.empty, Map[String,Seq[String]]())
+          HttpResponse(NOT_FOUND, JsArray.empty, Map[String,Seq[String]]())
         }
+      case _ => handleResponseEither(method, url)(response) match {
```

Comment:
`handleResponse` is now deprecated.

`handleResponseEither` has different behaviour for `400` and `404` responses which was the motivation for adding the it:tests and including the acceptance tests results on this pr.